### PR TITLE
fsmonitor: don't fill bitmap with entries to be removed

### PR DESCRIPTION
**Pull request by wilbaker**
This is the third iteration of changes to fix the segfault that I encountered while testing fsmonitor.  This iteration includes the following updates for feedback I received on v2:

- Update the new test case to use its own dedicated test repository

This latest v3 series has been reviewed by Dscho.

Thanks,
William

Cc: williamtbakeremail@gmail.com, stolee@gmail.com, Johannes.Schindelin@gmx.de, jeffhost@microsoft.com

### CODE REVIEW

**Comment by dscho** [comment]
Minor suggestions about the commit message:

- I usually try to break it up into multiple paragraphs, roughly following [Alan Alda's advice how to tell stories](https://www.youtube.com/watch?v=r4k6Gm4tlXw): give -- very briefly -- some background, then describe the obstacle, describe the various ways I attempted to solve it, then the solution. (Sometimes I demote paragraphs to the bottom that describe failed attempts.) Maybe you want to give it a try?
- In this instance, I think a very important thing to talk about is what you and me discussed yesterday: what if `fill_fsmonitor_dirty()` was called closer to the point where the index extension is written? ("Ah no, that did not work, because [... split index ...]".)

**Comment by wilbaker** [comment]
@dscho thanks for the feedback on the commit message, please let me know what you think of the updated commit message in my latest push.

**Comment by dscho** [comment]
> please let me know what you think of the updated commit message in my latest push.

@wilbaker I think it is great! Hopefully the Git maintainer will agree (they usually insist on a more "imperative tone", i.e. pretending that the commit tells Git to do this or that)...

File: t/t7519-status-fsmonitor.sh [hunk 1]
```diff
@@ -354,4 +354,21 @@ test_expect_success 'discard_index() also discards fsmonitor info' '
...
 '
 
+# Test staging/unstaging files that appear at the end of the index.  Test
+# files' names begin with 'z' so that they are sorted to the end of the index. 
+test_expect_success 'status succeeds after staging/unstaging ' '
+	test_create_repo fsmonitor-stage-unstage &&
+	(
+		cd fsmonitor-stage-unstage &&
+		test_commit initial &&
+		git update-index --fsmonitor &&
```
Comment (wilbaker) [inline t/t7519-status-fsmonitor.sh]: Interestingly, without this line:

```
git update-index --fsmonitor &&
```

This test would only catch the bug when run with `--run=29`.  If the entire script were run the issue did not repro (event with the new `BUG` statements in this patch).

**Pull request closed without merging**